pkg/controller/installation/installation_controller.go

```diff
@@ -138,6 +129,41 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		instance.Status.Stages = map[string]*v1alpha1.InstallationStageStatus{}
 	}
 
+	// If the CR is being deleted, cancel the current context
+	// and attempt to clean up the products with finalizers
+	if instance.DeletionTimestamp != nil {
+		// Cancel this context to kill all ongoing requests to the API
+		// and use a new context to handle deletion logic
+		r.cancel()
+
+		// Clean up the products which have finalizers associated to them
+		var merr error
```

Comment:
Is there a problem with initialising `merr` here so we don't need to perform the `nil` check each time we want to add to it. We can avoid the repeated casting too. Then we'd just need to check the length of the `errors` list in `merr` before deciding whether to return it or `nil` at the end.

Reply:
:+1: